janus_core/geo_opt.py

```diff
@@ -0,0 +1,55 @@
+"""Geometry optimisation."""
+
+from ase import Atoms
+
+try:
+    from ase.filters import FrechetCellFilter as DefaultFilter
+except ImportError:
+    from ase.constraints import ExpCellFilter as DefaultFilter
+
+from ase.optimize import LBFGS
+
+
+#  pylint: disable=dangerous-default-value
+#  Dictionaries are not modified within the function.
```

Comment:
I would still prefer:
```python
, dyn_kwargs = None, ...
```

with:
```python
dyn_kwargs = dyn_kwargs or {}
```
or
```python
dyn_kwargs = dyn_kwargs if dyn_kwargs else {}
```
rather than disabling the warning

Also, I believe this is disabling file-wide which you certainly don't want. (c.f. https://pylint.pycqa.org/en/latest/user_guide/messages/message_control.html)

Reply:
I was on the fence about this, as it feels unnecessarily verbose, but you're right, it's probably better to be safe. Should be fixed now.